src/engraving/libmscore/score.h

```diff
@@ -315,6 +315,25 @@ class UpdateState
     std::list<EngravingObject*> _deleteList;
 };
 
+template<typename T>
+class PaddingVector
+{
+private:
+    std::array<T, TOT_ELEMENT_TYPES> _paddingVector;
```

Comment:
The main thing I need here is to overload the `operator []` and the `.at()` members, cause I want to maintain the ability to access the table as `paddingTable[ElementType::type1][ElementType::type2]` (which is quite elegant I think, and I've used it all over the the codebase at this point). The overload needs to be explicitely defined cause the enum type isn't implicitely converted to its underlying int (I've tried).

Reply:
Ah, I see, that's what I had missed :)

Reply:
Do we really need to inherit something or add our own containers? Can we just use something like std::unordered_map instead?

Reply:
The current implementation _is_ an std::map. The type `PaddingTable` is currently defined as: 

`using PaddingTable = std::map<ElementType, <std::map<ElementType, double>>`.

This is elegant code-wise, because I can access the map by: 

`_paddingTable[ElementType::type1][ElementType::type2]`

(and at this point it is used everywhere in the codebase, so I would prefer not to change it). 
But reading from a map is much slower than reading from an array (and we are reading into this map thousands of times, so it makes a big difference). So the idea is that I redefine the PaddingTable as

`using PaddingTable = MyVector<MyVector<double>>`

`MyVector` is essentially built around an std::array, but I want to overload the `operator []` so I can still access the table by: 

`_paddingTable[ElementType::type1][ElementType::type2]`

That way, I don't need to change all the instances where the container is used and I keep it elegant, but it works much faster. Hope that makes sense :)

Reply:
using PaddingTable = std::unordered_map <ElementType, <std::unordered_map<ElementType, double>>

Reply:
Also, it is very interesting why we need to refill this table "tens of thousands of times"

Reply:
Every combination of [TypeA]-[TypeB] items has a minimum horizontal distance, specified by @oktophonie, which we call "padding". All paddings are very specific to each individual [TypeA]-[TypeB] combination (for example, BarLine-Note padding is different from BarLine-Accidental padding), and also they are not symmetric (for example, Note-BarLine padding is different from BarLine-Note padding). So we need a data structure to store these values, and it needs to be some form of 2D table. This table (i.e. the PaddingTable) is written only once, when the score is created, but is read a lot of times (every time one item has to decide the distance from the following item, it needs to read the appropriate padding value from the table). A big score can easily have tens of thousands of items, that's why I said the table could be accessed tens of thousands of times. 

So, I did some tests on a big score, and here's what I got.

Current master, which uses std::map:
<img width="665" alt="currentMaster" src="https://user-images.githubusercontent.com/93707756/193556619-9cd7bc30-035e-4895-823f-842d4d652c23.png">

Different implementation, using std::unordered_map (3.8% faster)
<img width="674" alt="unorderedMap" src="https://user-images.githubusercontent.com/93707756/193556783-fab086b6-b80e-4862-b298-8d50dcd6fb64.png">

This PR, using the custom container based on std::array (9.5% faster)
<img width="672" alt="thisPR" src="https://user-images.githubusercontent.com/93707756/193557100-7de971b3-8a49-497c-b25e-478fe0ebc999.png">